### PR TITLE
Optional extra arguments for custom error messages

### PR DESCRIPTION
**Pull request by onlinesid**
And unknown format no longer generate error

### CODE REVIEW

File: src/JsonSchema/Constraints/Constraint.php [hunk 1]
```diff
@@ -61,12 +61,20 @@ public function setUriRetriever(UriRetriever $uriRetriever)
     /**
      * {@inheritDoc}
      */
-    public function addError($path, $message)
+    public function addError($path, $message, $constraint='', $more=null)
     {
-        $this->errors[] = array(
+        $a = array(
```
Comment (keradus) [inline src/JsonSchema/Constraints/Constraint.php]: please don't use such a meaningless names

File: src/JsonSchema/Constraints/ConstraintInterface.php [hunk 2]
```diff
@@ -33,10 +33,12 @@ public function addErrors(array $errors);
...
      *
-     * @param $path
-     * @param $message
+     * @param string $path
+     * @param string $message
+     * @param string $constraint the constraint/rule that is broken, e.g.: 'minLength'
+     * @param array $more more array elements to add to the error
      */
-    public function addError($path, $message);
+    public function addError($path, $message, $constraint='', $more=null);
```
Comment (keradus) [inline src/JsonSchema/Constraints/ConstraintInterface.php]: if `$more` is an array it should be typehinted as `array`

File: src/JsonSchema/Constraints/Constraint.php (hunk 1 above)
Comment (onlinesid) [inline src/JsonSchema/Constraints/Constraint.php]: Hmm, it's better to use meaningless very short names for variables that are being used temporarily and locally within such a small method, but fine I'll change the name.

Comment (keradus) [inline src/JsonSchema/Constraints/Constraint.php]: you never know if this method will be larger in the future

**Comment by Maks3w** [comment]
Needs rebase

File: .gitignore [hunk 3]
```diff
@@ -7,3 +7,5 @@ coverage
 .settings
 composer.lock
 docs-api
+/.idea
+/composer.phar
```
Comment (bighappyface) [inline .gitignore]: Please move these to you [global ignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore)

**Comment by bighappyface** [comment]
@onlinesid thanks for the rebase. Would you please squash this down to a single commit?

**Comment by onlinesid** [comment]
I've squashed into a single commit now

**Comment by bighappyface** [comment]
@Maks3w what are your thoughts on the the version number to iterate to when this is merged?

**Pull request merged**